Bufwriter disassembly tweaks

ijackson opened the pull request:
In #80690 @BurntSushi observed that `WriterPanicked` was erroneously not exported, and suggested renaming `into_raw_parts` to `into_parts`. (More info in my commit messages.)

r? @BurntSushi

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/2721017090) [(plain)](https://github.com/rust-lang/rust/commit/49aef4b64ebd95f1b3316ca49c1faefe0c1a1b13/checks/2721017090/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.channel         := nightly
configure: rust.debug-assertions := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
configure: 
---
skip untracked path src/doc/rust-by-example/ during rustfmt invocations
skip untracked path src/llvm-project/ during rustfmt invocations
Diff in /checkout/library/std/src/io/mod.rs at line 262:
 
 #[stable(feature = "rust1", since = "1.0.0")]
 pub use self::buffered::IntoInnerError;
-#[stable(feature = "rust1", since = "1.0.0")]
-pub use self::buffered::{BufReader, BufWriter, LineWriter};
 #[unstable(feature = "bufwriter_into_raw_parts", issue = "80690")]
 pub use self::buffered::WriterPanicked;
+#[stable(feature = "rust1", since = "1.0.0")]
+pub use self::buffered::{BufReader, BufWriter, LineWriter};
 #[stable(feature = "rust1", since = "1.0.0")]
 pub use self::copy::copy;
 #[stable(feature = "rust1", since = "1.0.0")]
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustfmt" "--config-path" "/checkout" "--edition" "2018" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/collections/hash/mod.rs" "/checkout/library/std/src/collections/mod.rs" "/checkout/library/std/src/keyword_docs.rs" "/checkout/library/std/src/panic/tests.rs" "/checkout/library/std/src/io/prelude.rs" "/checkout/library/std/src/io/impls/tests.rs" "/checkout/library/std/src/io/mod.rs" "/checkout/library/std/src/time/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
Build completed unsuccessfully in 0:00:15
```

</details>

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-10`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/2721879256) [(plain)](https://github.com/rust-lang/rust/commit/1f6ba65470c352385188e4643395c173670dd17e/checks/2721879256/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- src/io/buffered/bufwriter.rs - io::buffered::bufwriter::BufWriter<W>::into_parts (line 326) stdout ----
error[E0658]: use of unstable library feature 'bufwriter_into_raw_parts'
  --> src/io/buffered/bufwriter.rs:334:48
   |
11 | let (recovered_writer, buffered_data) = stream.into_parts();
   |
   = note: see issue #80690 <https://github.com/rust-lang/rust/issues/80690> for more information
   = help: add `#![feature(bufwriter_into_raw_parts)]` to the crate attributes to enable

---

error: test failed, to rerun pass '--doc'


command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "--target" "x86_64-unknown-linux-gnu" "-Zbinary-dep-depinfo" "-j" "16" "--release" "--locked" "--color" "always" "--features" "panic-unwind backtrace compiler-builtins-c" "--manifest-path" "/checkout/library/test/Cargo.toml" "-p" "std" "--" "--quiet"


failed to run: /checkout/obj/build/bootstrap/debug/bootstrap --stage 2 test --exclude src/tools/tidy
Build completed unsuccessfully in 0:16:33
```

</details>

joshtriplett commented on library/std/src/io/buffered/mod.rs:
```diff
@@ -14,6 +14,8 @@ use crate::io::Error;
 
 pub use bufreader::BufReader;
 pub use bufwriter::BufWriter;
+#[unstable(feature = "bufwriter_into_raw_parts", issue = "80690")]
```
The feature gate name should change, as well.

ijackson replied:
Well, perhaps, in theory, but that seems unnecessarily awkward to me.  In #80690 it looks like this is going ot be stabilised imminently, so changing the feature gate name will just lead to churn.

But if you say it should still be changed, despite that consideration, I'll do that.  Thanks.

ijackson replied:
This seemed to be becoming a blocker so I have renamed it.

ijackson commented:
Rebased onto curret master.

joshtriplett commented:
Thanks for implementing the rename!

@bors r+

bors commented:
:pushpin: Commit 51d16ac7a840aecda2083585cf1003e3012a7576 has been approved by `joshtriplett`

<!-- @bors r=joshtriplett 51d16ac7a840aecda2083585cf1003e3012a7576 -->
<!-- homu: {"type":"Approved","sha":"51d16ac7a840aecda2083585cf1003e3012a7576","approver":"joshtriplett"} -->

JohnTitor commented:
Failed in rollup: https://github.com/rust-lang/rust/pull/87567#issuecomment-888619130
Needs a rebase to fix one more use on the latest master.
@bors r-

ijackson commented:
@JohnTitor Oh dear.  Sorry, this seems rather messed up  I'm a bit full of wine now :-) but I will look at it properly tomorrow.

ijackson commented:
Oh I see, 8837bf1acd2d0682c1030524fb328a073974112c introduced a use of `into_raw_parts` which then also needs to be renamed.

OK,  Rebased and fixed.  I also double checked the docs and they do look like everything is properly exported.

JohnTitor commented:
Thanks!
@bors r=joshtriplett

bors commented:
:pushpin: Commit bf30c51541c533830ec46ae65d9863666055a74b has been approved by `joshtriplett`

<!-- @bors r=joshtriplett bf30c51541c533830ec46ae65d9863666055a74b -->
<!-- homu: {"type":"Approved","sha":"bf30c51541c533830ec46ae65d9863666055a74b","approver":"joshtriplett"} -->

bors commented:
:hourglass: Testing commit bf30c51541c533830ec46ae65d9863666055a74b with merge 492723897e9b4db6701b3a75b72618d08a7d5319...
<!-- homu: {"type":"BuildStarted","head_sha":"bf30c51541c533830ec46ae65d9863666055a74b","merge_sha":"492723897e9b4db6701b3a75b72618d08a7d5319"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/3197157750)
Approved by: joshtriplett
Pushing 492723897e9b4db6701b3a75b72618d08a7d5319 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"joshtriplett","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/3197157750"},"merge_sha":"492723897e9b4db6701b3a75b72618d08a7d5319"} -->
